prespective_rectification/hough_transform_intersections.py

```diff
@@ -0,0 +1,78 @@
+import cv2
+import numpy as np
+from sklearn.cluster import KMeans
+
+from prespective_rectification.ploting import draw_lines
+from prespective_rectification.utils import convert_theta_to_two_points
+
+
+def hough_lines(image, cnts):
+    height, width, _ = image.shape
+
+    min_line_length = 10
+    max_line_gap = 10
+    lines = cv2.HoughLinesP(cnts, 1, np.pi / 180, 15, minLineLength=min_line_length, maxLineGap=max_line_gap)
+    clustering_parameters = []
+    clustering_parameters2 = []
+    for line in lines:
+        for x1, y1, x2, y2 in line:
+            if x2 == x1:
+                theta = np.sign(y2 - y1) * np.pi / 2
+            else:
+                p = (y2 - y1) / (x2 - x1)
+                theta = np.arctan(p)
+            x = (x1 + x2) * 0.5
+            y = (y1 + y2) * 0.5
+            clustering_parameters.append([x, y, theta * 100])
+            clustering_parameters2.append([x, y, theta])
+
+    startpts = gather_initial_points(image, clustering_parameters)
+    kmeans = KMeans(n_clusters=4, init=startpts, n_init=1)
+    kmeans.fit(clustering_parameters)
+    averaged_lines = average_lines_clustered(clustering_parameters2, kmeans.labels_)
+    draw_lines(image, averaged_lines, type_line="theta")
+    out_lines = []
+    for line in averaged_lines:
+        x, y, theta = line
+        out_lines.append(convert_theta_to_two_points(x, y, theta))
```

Comment:
Can be done in one line:
```suggestion
    out_lines = [convert_theta_to_two_points(x, y, theta) for x, y, theta in averaged_lines]
```

Reply:
way clear and clean that way yeah